iree/compiler/Dialect/HAL/Target/LLVM/internal/UnixLinkerTool.cpp

```diff
@@ -28,7 +28,7 @@ class UnixLinkerTool : public LinkerTool {
     if (!toolPath.empty()) return toolPath;
 
     // No explicit linker specified, search the environment for common tools.
-    toolPath = findToolInEnvironment({"ld", "ld.gold", "ld.lld"});
+    toolPath = findToolInEnvironment({"ld.lld", "ld.gold", "ld"});
```

Comment:
Can you elaborate on what this change helps with? (e.g. what path was discovered on your machine before, what path you want it to discover instead)

`ld.lld` is probably a good default here in UnixLinkerTool based on
```
lld is a generic driver.
Invoke ld.lld (Unix), ld64.lld (macOS), lld-link (Windows), wasm-ld (WebAssembly) instead
```
but I want to clarify a few details

The way the search is performed today might not be what you want with a simple reordering here: https://github.com/google/iree/blob/45bce4068608f047a4a1c8878828ff7d4f7331a5/iree/compiler/Dialect/HAL/Target/LLVM/LinkerTool.cpp#L200-L226

1. Search for _any_ matching tool in the executable's directory
2. Search for _any_ matching tool in the build tree
3. Search for _any_ matching tool in $PATH

You might instead want

1. Search for `toolNames[0]` in the exe's directory, the build tree, or $PATH
2. Search for `toolNames[1]` in the exe's directory, the build tree, or $PATH
3. Search for `toolNames[2]` in the exe's directory, the build tree, or $PATH

(You can also just be explicit if you have a use for a specific tool)

Reply:
(my task here is done. i'll cancel auto merge and let the experts bang it out 😁 )